zanata-war/src/main/java/org/zanata/action/VersionHomeAction.java

```diff
@@ -734,18 +740,22 @@ public void uploadSourceFile() {
             Optional<String> docType =
                 Optional.fromNullable(sourceFileUpload.documentType);
 
-            DocumentType documentType = getDocumentType(docType);
-
-            if (translationFileServiceImpl.hasAdapterFor(documentType)) {
-                uploadAdapterFile(documentType);
+            Optional<DocumentType> documentTypeOpt = getDocumentType(docType);
+            if (documentTypeOpt.isPresent()
+                    && translationFileServiceImpl.hasAdapterFor(documentTypeOpt
+                            .get())) {
+                uploadAdapterFile(documentTypeOpt.get());
+                resetPageData();
             } else {
+                String summary = "Unrecognized file extension for "
+                        +
+                        sourceFileUpload.getFileName();
+                // TODO this message is not displayed
                 conversationScopeMessages.setMessage(
```

Comment:
It should. But not in this PR.

On Wed, Sep 23, 2015 at 1:53 PM, Alex Eng <notifications@github.com> wrote:

> In zanata-war/src/main/java/org/zanata/action/VersionHomeAction.java
> <https://github.com/zanata/zanata-server/pull/928#discussion_r40169265>:
>
> >              } else {
> > +                String summary = "Unrecognized file extension for "
> > +                        +
> > +                        sourceFileUpload.getFileName();
> > +                // TODO this message is not displayed
> >                  conversationScopeMessages.setMessage(
>
> Should conversationScopeMessages gets replace?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/928/files#r40169265>.
>



-- 
Patrick Huang
Senior Software Engineer
Engineering - Internationalisation
Red Hat, Asia-Pacific Pty Ltd
Level 1, 193 North Quay
Brisbane 4000
Office: +61 7 3514 8278
Fax: +61 7 3514 8199
IRC: pahuang
github: github.com/huangp
Website: www.redhat.com